src/main/utils/resource.ts

```diff
@@ -1,112 +1,112 @@
+        private _homeDir: string,
+        private downloadSource: string,
+        resourceName: string,
+        private _logManager: LogManager,
+        connectionManager?: JfrogClient
+    ) {
+        this.downloadDir = path.join(_homeDir, 'download');
+        this.downloadTarget = path.join(this.downloadDir, resourceName);
+        this.path = path.join(_homeDir, resourceName);
+        if (connectionManager !== undefined) {
+            // Override the default connection manager.
+            this._connectionManager = connectionManager;
+        }
+    }
 
-//             await this.copy(path.join(this.TMP_DOWNLOAD_PATH, this.resourceName), path.join(this.resourceLocalPath, this.resourceName));
-//         } catch (error) {
-//             this._logManager.logMessage('failed to update the applicable scanner: ' + error, 'WARN');
-//         } finally {
-//             fs.rmSync(this.TMP_DOWNLOAD_PATH, { recursive: true, force: true });
-//         }
-//     }
-//     public copy(oldPath: string, newPath: string) {
-//         return new Promise((resolve, reject) => {
-//             const readStream: fs.ReadStream = fs.createReadStream(oldPath);
-//             const writeStream: fs.WriteStream = fs.createWriteStream(newPath);
+    public get homeDir(): string {
+        return this._homeDir;
+    }
 
-//             readStream.on('error', err => reject(err));
-//             writeStream.on('error', err => reject(err));
+    // Update the resource to the latest released
+    public async update(withExecPrem: boolean) {
+        const updateAvailable: boolean = await this.isUpdateAvailable();
+        if (!updateAvailable) {
+            return;
+        }
+        this._logManager.logMessage('Downloading new update from ' + this.downloadTarget, 'DEBUG');
+        if (this.isUpdateStarted()) {
+            if (this.isUpdateStuck() === false) {
+                return;
+            }
+            fs.rmSync(this.downloadDir, { recursive: true });
+        }
+        fs.mkdirSync(this.downloadDir, { recursive: true });
+        try {
+            await this._connectionManager
+                .artifactory()
+                .download()
+                .downloadArtifactToFile(this.downloadSource, this.downloadTarget);
+            if (withExecPrem) {
+                fs.chmodSync(this.downloadTarget, '755');
+            }
+            fs.copyFileSync(this.downloadTarget, this.path);
+            this._logManager.logMessage('Update resource was successfully upgraded for ' + this.downloadSource, 'DEBUG');
+        } finally {
+            fs.rmSync(this.downloadDir, { recursive: true, force: true });
+        }
+    }
 
-//             writeStream.on('close', function() {
-//                 resolve(true);
-//             });
+    public async isUpdateAvailable(): Promise<boolean> {
+        if (!fs.existsSync(this.path)) {
+            return true;
+        }
+        let checksumResult: IChecksumResult = { sha256: '', sha1: '', md5: '' };
```

Comment:
I think that sha1 + md5 are not really needed.
```suggestion
        let checksumResult: IChecksumResult = { sha256: '' };
```

Reply:
The following compile error is occurring:
```
Type '{ sha256: string; }' is missing the following properties from type 'IChecksumResult': sha1, md5
```